.travis.yml

```diff
@@ -0,0 +1,49 @@
+dist: trusty
+sudo: off
+
+language: node_js # Base language is Nodejs (used for liniting and basic validation).
+
+addons:
+  apt:
+    sources:
+    - sourceline: 'deb https://packages.microsoft.com/repos/azure-cli/ trusty main'
+      key_url: 'https://packages.microsoft.com/keys/microsoft.asc'
+    packages:
+      - azure-cli # Install the Azure CLI tools
+
+before_script:
+  - npm install -g grunt-cli # Install task runners for lint checking.
+
+node_js:
+  - "0.12"
+
+cache:
+  directories:
+    - node_modules
+
+script: # Run the lint tests, then if env vars are present validate the template then try building the stack.
+  - npm test
+  - if [ "$SPNAME" != ""  -a  "$SPPASSWORD" != ""  -a  "$SPTENANT" != ""  -a  "$SPSSHKEY" != "" ]; then
+      echo "Running Azure setup steps.";
+      az login --service-principal -u "$SPNAME" -p "$SPPASSWORD" --tenant "$SPTENANT";
+      az group create -l westus -g "moodle-azure-$TRAVIS_BUILD_NUMBER";
+    else
+      echo "Skipping Azure setup.";
+    fi
+  - if [ "$SPNAME" != ""  -a  "$SPPASSWORD" != ""  -a  "$SPTENANT" != ""  -a  "$SPSSHKEY" != "" ]; then
+      echo "Running Azure validation step.";
+      az group deployment validate --resource-group "moodle-azure-$TRAVIS_BUILD_NUMBER" --template-file azuredeploy.json --parameters azuredeploy.parameters.json --parameters sshPublicKey="$SPSSHKEY";
```

Comment:
Are we able to remove this as a dependency all together? It looks like in previous versions there was a generic value applied. Do we know how this worked?

Reply:
I'm not sure about the previous versions and a generic value you are mentioning here, but thinking about this again, it looks like we better have this supplied from outside, rather than internally generating, because I'm not sure how the internally generated SSH priv key can be made available to us later (it may not be required, though). I guess we can also change this behavior later, like the LOCATION variable above, so let's forget about this for now. Thanks!